packages/curve-ui-kit/src/lib/model/query/factory.ts

```diff
@@ -1,21 +1,13 @@
 import { CB } from 'vest-utils'
-import { QueryFunctionContext, queryOptions } from '@tanstack/react-query'
+import { QueryFunctionContext, queryOptions, useQuery } from '@tanstack/react-query'
 import { queryClient } from '@ui-kit/lib/api/query-client'
 import { logQuery } from '@ui-kit/lib/logging'
 import { REFRESH_INTERVAL } from '@ui-kit/lib/model/time'
-import { createQueryHook } from '@ui-kit/lib/queries'
 import { QueryFactoryInput, QueryFactoryOutput } from '@ui-kit/lib/types'
 import { assertValidity as sharedAssertValidity, checkValidity, FieldName, FieldsOf } from '@ui-kit/lib/validation'
 
-export function getParamsFromQueryKey<TKey extends readonly unknown[], TParams, TQuery, TField>(
-  queryKey: TKey,
-  assertValidity: (data: TParams, fields?: TField[]) => TQuery,
-) {
-  const queryParams = Object.fromEntries(
-    queryKey.flatMap((i) => (i && typeof i === 'object' ? Object.entries(i) : [])),
-  ) as TParams
-  return assertValidity(queryParams)
-}
+const getParamsFromQueryKey = <TKey extends readonly unknown[], TParams, TQuery, TField>(queryKey: TKey) =>
```

Comment:
Doesn't have to be exported as there's only one usage (down below), so I removed it